src/main/java/Box.java

```diff
@@ -0,0 +1,57 @@
+import java.util.ArrayList;
+import java.util.Collections;
+
+/**
+ * Box
+ */
+public class Box<T extends Fruit<T>> {
+    protected final ArrayList<Fruit<T>> fruits = new ArrayList<>();
+
+    /**
+     * @param fruit (Fruit<T> fruit)
+     */
+    public void add(Fruit<T> fruit) {
+        fruits.add(fruit);
+    }
+
+    /**
+     * @param fruit (Fruit<T>[] fruit)
+     */
+    public void add(Fruit<T>[] fruit) {
+        Collections.addAll(fruits, fruit);
+    }
+
+    public float getWeight() {
+        var weight = 0.0F;
+        for (var fruit : fruits) {
+            weight += fruit.weight;
+        }
+        return weight;
+    }
+
+    /**
+     * @param box1 (Box<T> box)
+     * @param box2 (Box<T> box)
+     */
+    public boolean compare(Box<?> box1, Box<?> box2) {
```

Comment:
`Box<? extends Fruit>`
Currently, you may have in the box other types. For example - bricks